src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -313,7 +313,7 @@ private void UpdateCurrentConfiguration()
             }
             finally
             {
-                _syncAfter = DateTimeUtil.Add(DateTime.UtcNow, AutomaticRefreshInterval < RefreshInterval ? AutomaticRefreshInterval : RefreshInterval);
+                _syncAfter = DateTimeUtil.Add(DateTime.UtcNow, AutomaticRefreshInterval);
                 Interlocked.Exchange(ref _configurationRetrieverState, ConfigurationRetrieverIdle);
```

Comment:
We don't seem to set `_configurationRetrieverState` to Running anywhere?

Reply:
It looks like it is set to running just before this method gets started in a new task. Or does the fact that it's in a new task change the call context?

Reply:
On ln 251? It's being set to ` ConfigurationRetrieverIdle`. Regardless, I think we should address it outside of this PR

Reply:
Oh, so the arguments are flipped in the call to `Interlocked.CompareExchange`.

Reply:
Yes, I think so. (The CompareExchange is not super intuitive since it compares the first and third parameters...)
```csharp
if (Interlocked.CompareExchange(ref _configurationRetrieverState, ConfigurationRetrieverRunning, ConfigurationRetrieverIdle) != ConfigurationRetrieverRunning)
                {
                    _ = Task.Run(UpdateCurrentConfiguration, CancellationToken.None);
                }
```
Logic:
If the state is idle (is not running),
   then set state to running and call the update
If the state is running
   then don't do anything.

at the end of Update method, we correctly reset the state to idle https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/e8e9ecfe074b272e4c101b8865ee63409f59b400/src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs#L317